### PR TITLE
Unwrap "double errors" to limit cost of Error ctor

### LINKED ISSUE #14367: Unwrap "double errors" to limit cost of Error ctor
# Bug report

Related to #13532

**What is the current behavior?**
As discussed in the previous issue, new-ing up Errors has a high cost. This is especially relevant when using a bunch of type-only imports in typescript. Each of these issues causes a `HarmonyLinkingError`, which is then immediately wrapped in a `ModuleDependencyWarning`. This double-wrapping causes additional overhead due to both of them extending `Error` and paying the price of that.

**What is the expected behavior?**
My suggestion is to cut out the usage  `ModuleDependency{Warning,Error}` and directly use the errors returned from `getWarnings`. I have looked through the codebase and dont see anywhere that specifically cares about `ModuleDependencyXXX` error type over other types of WebpackErrors, so this seems like a safe change to make.

See cpu trace images below:

*Before*
image

*After*
image

In our very large typescript-based project webpack build, this change saves ~200ms per recompile. Attaching changes.

**Other relevant information:**
webpack version: 5.55.1
Node.js version: 16.9.0
Operating System: windows
Additional tools: -

### PR DESCRIPTION
**Pull request by markjm**
Fixes #14367

**What kind of change does this PR introduce?**
performance

**Did you add tests for your changes?**
no

**Does this PR introduce a breaking change?**
no

**What needs to be documented once your changes are merged?**
none

### CODE REVIEW

File: lib/util/internalSerializables.js [hunk 1]
```diff
@@ -169,7 +169,7 @@ module.exports = {
 		require("../InvalidDependenciesModuleWarning"),
 	Module: () => require("../Module"),
 	ModuleBuildError: () => require("../ModuleBuildError"),
-	ModuleDependencyWarning: () => require("../ModuleDependencyWarning"),
+	HarmonyLinkingError: () => require("../HarmonyLinkingError"),
```
Comment (markjm) [inline lib/util/internalSerializables.js:172]: Dont quite understand this, so not sure if this is required

Comment (sokra) [reply lib/util/internalSerializables.js:172]: Actually it's not. Neither `ModuleDependencyWarning` nor `HarmonyLinkingError` are serialized, so both could be omitted here. And you also need to omit the `makeSerializable` in the class.

Comment (markjm) [reply lib/util/internalSerializables.js]: removed from this file. HarmonyLinkingError class did not previously have makeSerializable, so nothing to remove i think

**Comment by kenotron** [comment]
Hey @sokra - can you check this one again?

**Comment by sokra** [comment]
> Hey @sokra - can you check this one again?

We instead merged another PR tackling the same problem: https://github.com/webpack/webpack/pull/14404

This allows to avoid creating these errors instead of creating them and ignoring them again...

**Pull request closed without merging**